test/unit/BNum.tree

```diff
@@ -0,0 +1,97 @@
+BNumTest::btoi
+├── when passing zero
+│   └── it should return zero
+├── when passing BONE
+│   └── it should return one
+├── when passing a value less than BONE
+│   └── it should return zero
+└── when using known values
+    └── it should return correct value
+
+
+BNumTest::bfloor
+├── when passing zero
+│   └── it should return zero
+├── when passing a value less than BONE
+│   └── it should return zero
+└── when using known values
+    └── it should return correct value
+
+
+BNumTest::badd
+├── when passing zero and zero
+│   └── it should return zero
+├── when passing a as uint256 max and b non zero
+│   └── it should revert
+├── when passing b as uint256 max and a non zero
+│   └── it should revert
+└── when passing known values
+    └── it should return correct value
+
+
+BNumTest::bsub
+├── when passing zero and zero
+│   └── it should return zero
+├── when passing a less than b
+│   └── it should revert
+└── when passing known values
+    └── it should return correct value
+
+
+BNumTest::bsubSign
+├── when passing zero and zero
+│   └── it should return zero and false
+├── when passing a less than b
+│   └── it should return correct value and true
+└── when passing known values
+    └── it should return correct value
+
+
+BNumTest::bmul
+├── when passing zero and zero
+│   └── it should return zero
+├── when passing a as uint256 max and b non zero
+│   └── it should revert
+├── when passing b as uint256 max and a non zero
+│   └── it should revert
+├── when passing a mul b too big
+│   └── it should revert
+│       └── a * b + BONE / 2 > uint256 max
+└── when passing known values
+    └── it should return correct value
+
+
+BNumTest::bdiv
+├── when passing b as zero
+│   └── it should revert
+├── when passing a as uint256 max
+│   └── it should revert
+├── when passing a and b too big
+│   └── it should revert
+│       └── a*BONE/b + b/2 > uint256 max
+├── when flooring to zero
+│   └── it should return 0
+└── when passing known values
+    └── it should return correct value
+
+
+BNumTest::bpowi
```

Comment:
opinion on having a revert when a number with BONE precision is passed as the exponent? other functions already have 'reverts on overflows caused by parameters' covered

Reply:
hmmm i don't think that's actually a branch, but rather a result. gave it a try, and appears that even `(BONE + 1 )^ BONE` is actually possible (being `BONE+1` the minimum base to have an effect) to calculate, but `(BONE + 1)^ 100 * BONE` (arbitrary range) does always overflow (according to foundry fuzzer).

i believe this is more task of echidna than BTT